src/ProjectNeon/Assets/Features/Effects/Timed.cs

```diff
@@ -0,0 +1,37 @@
+﻿
+/**
+ * Wraps the execution of an effect so it is only executed for a certain amount of turns.
+ */
+public class Timed : ITemporalState, Effect
```

Comment:
What about a name like `ForNumberOfTurns`? Perhaps that would make the code more clear at usage sites.